src/java/org/apache/cassandra/cql3/constraints/ColumnConstraint.java

```diff
@@ -42,10 +69,17 @@ public interface ColumnConstraint
     void evaluate(Class<? extends AbstractType> valueType, Object columnValue) throws ConstraintViolationException;
```

Comment:
This can be changed to 
```suggestion
    void evaluate(AbstractType<?> valueType, ByteBuffer columnValue) throws ConstraintViolationException;
```